src/lib/azure-devops/azure-devops-contributors.ts

```diff
@@ -0,0 +1,152 @@
+    }
+    } else {
+      // Otherwise retrieve all repos (for given projects or all repos)
+      repoList = repoList.concat(
+        await fetchAzureReposForProjects(azureInfo)
+      );
+    }
+
+    if(SnykMonitoredRepos && SnykMonitoredRepos.length > 0){
+      repoList = repoList.filter(repo => SnykMonitoredRepos.includes(`${repo.project.key}/${repo.name}`) || SnykMonitoredRepos.includes(`${repo.project.name}/${repo.name}`))
+    }
+
+
+    for (let i = 0; i < repoList.length; i++) {
+      await fetchAzureContributorsForRepo(
+        azureInfo,
+        repoList[i],
+        contributorsMap,
+        threeMonthsDate
+      );
+    }
+  } catch (err) {
+    debug("Failed to retrieve contributors from Azure Devops.\n" + err);
+    console.log(
+      "Failed to retrieve contributors from Azure Devops. Try running with `DEBUG=snyk* snyk-contributor`"
+    );
+     }
+    debug(contributorsMap);
+    return contributorsMap;
+};
+
+export const fetchAzureContributorsForRepo = async (
+  AzureInfo: AzureDevopsTarget,
+  repo: Repo,
+  contributorsMap: ContributorMap,
+  threeMonthsDate: string
+): Promise<void> => {
+  try {
+    debug(
+      `Fetching single repo contributor from Azure Devops. Project ${repo.project.key} - Repo ${repo.name}\n`
+    );
+    const response = (await getRepoCommits(
+      azureDefaultUrl + AzureInfo.OrgName,
+      repo.project.key,
+      repo.name,
+      AzureInfo.token,
+      threeMonthsDate
+    ));
+    const result = await response.text();
+    const parsedResponse = JSON.parse(result).value as Commits[];
+    for (let i = 0; i < parsedResponse.length; i++) {
+      const commit = parsedResponse[i];
+      let contributionsCount = 1;
+      let reposContributedTo = [`${repo.project.name || repo.project.key}/${repo.name}`];
+
+      if (contributorsMap && contributorsMap.has(commit.author.name)) {
+        contributionsCount =
+          contributorsMap.get(commit.author.name)?.contributionsCount || 0;
+        contributionsCount++;
+
+        reposContributedTo =
+          contributorsMap.get(commit.author.name)?.reposContributedTo || [];
+        if (!reposContributedTo.includes(`${repo.project.name || repo.project.key}/${repo.name}`)) {
+          // Dedupping repo list here
+          reposContributedTo.push(`${repo.project.name}/${repo.name}`);
+        }
+      }
+      contributorsMap.set(commit.author.name, {
+        email: commit.author.email,
+        contributionsCount: contributionsCount,
+        reposContributedTo: reposContributedTo,
+      });
+    }
+  } catch (err) {
+    debug("Failed to retrieve commits from Azure Devops.\n" + err);
+    console.log(
+      "Failed to retrieve commits from Azure Devops. Try running with `DEBUG=snyk* snyk-contributor`"
+    );
+  }
+};
+
+export const fetchAzureReposForProjects = async (
+  AzureInfo: AzureDevopsTarget
+): Promise<Repo[]> => {
+  const repoList: Repo[] = [];
+  if(AzureInfo.projectKeys){
+  try {
```

Comment:
indentation is wrong here, makes it hard to read. if there are no projectKeys, it's just returning nothing? should probably be more of an error or something. Probably addressed by making projectKeys required ?

Reply:
Yes, chamged to required